### PR TITLE
feat(sdk-coin-atom): skeleton for transaction building and account-lib ATOM

### PR DESCRIPTION
**Pull request by michael-gallegos-bg**
Adds basic functionality for decoding transactions for ATOM transfers.
Also added skeleton for transaction builder factory, transfer builder, and account-lib exposure.

Will be working on adding additional testing in follow up PRs, working out skeleton right now to unblock account-lib KeyPair access on the WP side.

TICKET: BG-64363

### CODE REVIEW

File: modules/sdk-coin-atom/src/lib/iface.ts
Comment (michael-gallegos-bg) [inline modules/sdk-coin-atom/src/lib/iface.ts:26]: Need to refactor this in the future to rename the fields if we wish

**Review by sachushaji — DISMISSED** [review]
Package.json and lock changes looks good to me

File: modules/sdk-coin-atom/src/lib/iface.ts
Comment (Vijay-Jagannathan) [inline modules/sdk-coin-atom/src/lib/iface.ts:19]: Based on our sandbox scripts, `gasLimit` seems like a `string`

File: modules/sdk-coin-atom/src/lib/transaction.ts
Comment (Vijay-Jagannathan) [inline modules/sdk-coin-atom/src/lib/transaction.ts:27]: Would be great if we had something like [Lombok](https://projectlombok.org/features/GetterSetter) in typescript to reduce boilerplate and look easy on the eye, but I'm not sure if there's a decent library that does that

File: modules/sdk-coin-atom/src/lib/transaction.ts
Comment (Vijay-Jagannathan) [inline modules/sdk-coin-atom/src/lib/transaction.ts:180]: More of a question for my understanding since I'm new to typescript:

I see that we've initialized a class variable called `_atomTransaction` with a leading underscore to indicate that it's private. However, in line 180 and 183, I see that underscore is not present. So my question is whether this is interchangeable or a typo ? If it's interchangeable, could we stick to one way of referencing the class variable (either with _ or the regular way) ?

File: modules/sdk-coin-atom/src/lib/transaction.ts
Comment (Vijay-Jagannathan) [inline modules/sdk-coin-atom/src/lib/transaction.ts:52]: Is this temporary or is it always supposed to return true ?

File: modules/sdk-coin-atom/src/lib/transaction.ts
Comment (Vijay-Jagannathan) [inline modules/sdk-coin-atom/src/lib/transaction.ts:145]: Could we have a helper method which does this `MessageData` construction for us instead of the `deseralizeAtomTransaction` method doing it ?

File: modules/sdk-coin-atom/src/lib/transaction.ts
Comment (Vijay-Jagannathan) [inline modules/sdk-coin-atom/src/lib/transaction.ts:161]: Similar to the comment in the interface file

File: modules/sdk-coin-atom/src/lib/iface.ts
Comment (michael-gallegos-bg) [reply modules/sdk-coin-atom/src/lib/iface.ts:19]: I think it might be getting converted in their SDK, because in my debugging when deserializing from an encoded base64 string, it came back as a number.

File: modules/sdk-coin-atom/src/lib/transaction.ts
Comment (michael-gallegos-bg) [reply modules/sdk-coin-atom/src/lib/transaction.ts:27]: This will be abstracted more anyways via the transaction builder and the factory.

File: modules/sdk-coin-atom/src/lib/transaction.ts
Comment (michael-gallegos-bg) [reply modules/sdk-coin-atom/src/lib/transaction.ts:145]: Yes, this was just an initial draft, will be modularizing it today.

File: modules/sdk-coin-atom/src/lib/transaction.ts
Comment (michael-gallegos-bg) [reply modules/sdk-coin-atom/src/lib/transaction.ts:161]: From my comment before, it was coming back as a Long, so I cast it to a number type, but we can keep it as a long or even convert to a string if we wish.

File: modules/sdk-coin-atom/src/lib/transaction.ts
Comment (Vijay-Jagannathan) [reply modules/sdk-coin-atom/src/lib/transaction.ts:180]: Ah gotcha, I read the `get atomTransaction()` as `getAtomTransaction()`. Now it makes sense

File: modules/sdk-coin-atom/src/lib/transaction.ts
Comment (Vijay-Jagannathan) [inline modules/sdk-coin-atom/src/lib/transaction.ts:31]: Based on the getter method, it seems like `set atomTransaction(...)` would be better in terms of consistency. Not a blocking change though, can be addressed in a follow up.

File: modules/sdk-coin-atom/src/lib/transaction.ts
Comment (Vijay-Jagannathan) [inline modules/sdk-coin-atom/src/lib/transaction.ts:86]: Question: It does seem like this `TransactionExplanation` object is constructed in such a way that there is inherent ordering as per the `displayOrder` constant. Do we still need the constant for some other purpose ?

**Review by sachushaji — COMMENTED** [review]
Flushing review queue, Re-request when ready

File: modules/sdk-coin-atom/src/lib/iface.ts
Comment (dpkjnr) [reply modules/sdk-coin-atom/src/lib/iface.ts:19]: if it came back as long then BigInt would be better option I think. JS number type is notorious for precision issues

File: modules/sdk-coin-atom/src/lib/transaction.ts
Comment (michael-gallegos-bg) [reply modules/sdk-coin-atom/src/lib/transaction.ts:86]: I'm not entirely sure, we can ask the greater team and see if we can get more insight, but the constant is present for all other coins' implementation of this specific method.

File: modules/sdk-coin-atom/test/unit/transaction.ts
Comment (michael-gallegos-bg) [inline modules/sdk-coin-atom/test/unit/transaction.ts:27]: Will be adding more testing in a follow up PR, right now getting the skeleton's out to get closer to unblocking wallet creation via account lib in WP

File: modules/sdk-coin-atom/src/lib/transferBuilder.ts
Comment (Vijay-Jagannathan) [inline modules/sdk-coin-atom/src/lib/transferBuilder.ts:5]: Question: Any specific reason why we have a separate `TransferBuilder` class to just do a get `transactionType` or is this how other coins are doing it ?

**Review by Vijay-Jagannathan — APPROVED** [review]

File: modules/sdk-coin-atom/src/lib/constants.ts
Comment (sachushaji) [inline modules/sdk-coin-atom/src/lib/constants.ts:2]: Whats the need of hardcoding a RPC endpoint here?

File: modules/sdk-coin-atom/src/lib/utils.ts
Comment (sachushaji) [inline modules/sdk-coin-atom/src/lib/utils.ts:172]: Can you assign ticket numbers for this TODOs?

File: modules/sdk-coin-atom/test/unit/transaction.ts
Comment (sachushaji) [reply modules/sdk-coin-atom/test/unit/transaction.ts:27]: Can you add ticket numbers for this follow ups

File: modules/sdk-coin-atom/src/lib/constants.ts
Comment (sachushaji) [inline modules/sdk-coin-atom/src/lib/constants.ts:2]: Whats the need of hardcoding a RPC endpoint here?

File: modules/sdk-coin-atom/src/lib/utils.ts
Comment (sachushaji) [inline modules/sdk-coin-atom/src/lib/utils.ts:172]: Can you assign ticket numbers for this TODOs?

File: modules/sdk-coin-atom/src/lib/utils.ts
Comment (sachushaji) [inline modules/sdk-coin-atom/src/lib/utils.ts:192]: Kindly remove Console statements

Comment (michael-gallegos-bg) [reply modules/sdk-coin-atom/src/lib/utils.ts:192]: Good call! Thank you

File: modules/sdk-coin-atom/src/lib/utils.ts
Comment (michael-gallegos-bg) [reply modules/sdk-coin-atom/src/lib/utils.ts:172]: Yes will do shortly

File: modules/sdk-coin-atom/src/lib/transferBuilder.ts
Comment (michael-gallegos-bg) [reply modules/sdk-coin-atom/src/lib/transferBuilder.ts:5]: I think when we get further along the line, we will need to decorate the transaction depending on the type we are building (transfer, staking, other)

And yes this is how other coins do this

**Review by sachushaji — APPROVED** [review]
Thanks for the hard work @michael-gallegos-bg. One thong to not is that the PR has only minimal tests for a lot of functionalities which you added. I think you can add more for the functions which you have implemented, But I see a follow up ticket mentioned in the test file for it, So I'm not blocking this PR.

File: modules/sdk-coin-atom/src/lib/utils.ts
Comment (michael-gallegos-bg) [reply modules/sdk-coin-atom/src/lib/utils.ts:69]: Yes this is a follow up ticket that we have, this is the next step we will be working on - just wanted to get this out in account-lib for now to unblock KeyPair access on the WP side. Thanks!

**Pull request merged**
